DiscordChatExporter.Core/Exporting/ExportAssetDownloader.cs

```diff
@@ -101,12 +102,32 @@ await using (var output = File.Create(filePath))
 
 internal partial class ExportAssetDownloader
 {
-    private static string GetUrlHash(string url) =>
-        SHA256
+    private static string GetUrlHash(string url)
+    {
+        // Strip out ex, is and hm query params from Discord CDN URLs to create a consistent hash
+        if (
+            url.Contains("cdn.discordapp.com")
+            && url.Contains("ex")
+            && url.Contains("is")
+            && url.Contains("hm")
+        )
+        {
+            var uriBuilder = new UriBuilder(url);
+            var query = HttpUtility.ParseQueryString(uriBuilder.Query);
+            query.Remove("ex");
+            query.Remove("is");
+            query.Remove("hm");
+            uriBuilder.Query = query.ToString();
+            url = uriBuilder.ToString();
+            url = url.Replace(":443", "");
```

Comment:
Is this required? 🤔

Reply:
```cs
url = url.Replace(":443", "");
```
this is needed, because `Uribuilder` for some reason adds a port to the url
`https://cdn.discordapp.com/attachments/...` -> `https://cdn.discordapp.com:443/attachments/...`

The goal is to transform the url to the format as the one before Discord added signed urls.